shm/Player.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#include <memory>
+
+#include "Ship.hpp"
+
+class Player {
+	std::unique_ptr<Ship> ship_;
+	int money_;
+	size_t availableSpace_;
+
+public:
+	Player():
+		ship_(std::make_unique<Ship>()),
```

Comment:
Maybe ship_ should stay nullptr? Until you add some to player?

Reply:
Also, you don't have a method to add Ship to the player. So why you allow Create `Player` object without specific `Ship` :)? The default one is bad I guess.